### PR TITLE
ci: implement renovate

### PR DESCRIPTION
**Pull request by strophy**
Manually updating dependencies is boring, and creates increasing amounts of work as we add support for more plugins. This PR implements [Renovate](https://github.com/renovatebot/renovate/) bot to update dependencies. It will be necessary to create a GitHub token named `RENOVATE_TOKEN` with repo level access to use this bot, as described [here](https://github.com/renovatebot/github-action#token). 

I have stored configuration metadata in `renovate.json`, but much of this could be moved into `deps.list` itself as annotations to each line, describing the type and source of the dependency update ([see example here](https://docs.renovatebot.com/modules/manager/regex/#advanced-capture)). This would be a bit more DRY, but makes the deps file look quite cluttered.

Since Docker Hub is actually the source of many of our updates, I have moved entire image tags into the deps.list file, to avoid problems such as `golang:${GO_VERSION}-alpine${ALPINE_VERSION}` being parsed into `golang:1.19.3-alpine3.17` when this image does not actually exist yet, because Alpine 3.17 is still very new.

Finally, I have not yet been able to get the bot to support updating the commit hash for `GOOGLE_API_VERSION`. Issue open with Renovate devs [here](https://github.com/renovatebot/renovate/discussions/19076).

### CODE REVIEW

**Review by rvolosatovs — COMMENTED** [review]
Nice work, excited to see this in action.
Please squash the `fix` commit with the parent commit

The quickest way is probably:
```
git reset HEAD^
git add .github
git commit --amend
```

You can use `git diff` to validate that everything is "still there", for example:
```
git diff upstream/ci/renovate
```
Assuming this repository is locally tracked as `upstream` this will give you a diff between what you have locally and the exact state of affairs in this PR

**Review by rvolosatovs — APPROVED** [review]

File: .github/workflows/renovate.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    - cron: '30 5 * * 0'
+
+jobs:
+  renovate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Self-hosted Renovate
```
Comment (rvolosatovs) [inline .github/workflows/renovate.yml]: It's not self-hosted. IMO we can just skip the `name` field

File: .github/workflows/renovate.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '30 5 * * 0'
+
+jobs:
+  renovate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
```
Comment (rvolosatovs) [inline .github/workflows/renovate.yml]:
```suggestion
```

File: .github/workflows/renovate.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: Renovate
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '30 5 * * 0'
+
+jobs:
+  renovate:
+    runs-on: ubuntu-latest
```
Comment (rvolosatovs) [inline .github/workflows/renovate.yml]: Should we version this just like in other workflows?

File: .github/workflows/renovate.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Self-hosted Renovate
+        uses: renovatebot/github-action@v34.42.0
+        with:
+          configurationFile: .github/renovate.json
+          token: ${{ secrets.RENOVATE_TOKEN }}
```
Comment (rvolosatovs) [inline .github/workflows/renovate.yml:17]: I added a token for now, but going forward let's switch to an app instead https://github.com/renovatebot/github-action#example-with-github-app

Comment (strophy) [reply .github/workflows/renovate.yml:17]: Sure, I'll swap it over to a GitHub App in another PR. Let's see if this config performs well first - you can see examples of the PRs it creates over here: https://github.com/strophy/docker-protobuf/pulls?q=is%3Apr+is%3Aclosed

**Pull request merged**